File: survey_sharing_backend/src/main/java/it/lorenzobloise/survey_sharing_backend/entities/Statistics.java
package it.lorenzobloise.survey_sharing_backend.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Version;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document("statistics")
@ToString
@EqualsAndHashCode
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Statistics {

    @Id
    private String id;
    @Indexed(unique = true)
    @JsonIgnore
    private String survey;
    private int numberOfAnswers;
    private double percentOfUsersWhoAnswered;
    private double averageAge;
    private List<Integer> ageList;
    private int numberOfMaleUsersWhoAnswered;
    private int numberOfFemaleUsersWhoAnswered;
    private List<String> countriesList;
    private int numberOfDifferentCountries;
    private int numberOfInvitationsSent;
    private double percentOfInvitationsAccepted;
    private List<List<String>> listOfFeedbacks;
    private int numberOfPositiveFeedbacks;
    private int numberOfNegativeFeedbacks;
    private int numberOfMixedFeedbacks;
    private int numberOfNeutralFeedbacks;
    private List<Double> ratings;
    private double averageRating;
    @JsonIgnore
    @Version
    private Long version;

}

File: survey-sharing-frontend/src/app/entities/statistics.ts
import { Survey } from "./survey";

export class Statistics{
  id!: string;
  survey!: string;
  numberOfAnswers!: number;
  percentOfUsersWhoAnswered!: number;
  averageAge!: number;
  ageList!: number[];
  numberOfMaleUsersWhoAnswered!: number;
  numberOfFemaleUsersWhoAnswered!: number;
  countriesList!: string[];
  numberOfDifferentCountries!: number;
  numberOfInvitationsSent!: number;
  percentOfInvitationsAccepted!: number;
  listOfFeedbacks!: string[][];
  numberOfPositiveFeedbacks!: number;
  numberOfNegativeFeedbacks!: number;
  numberOfMixedFeedbacks!: number;
  numberOfNeutralFeedbacks!: number;
  ratings!: number[];
  averageRating!: number;

  constructor(id: string, survey: string, numberOfAnswers: number, percentOfUsersWhoAnswered: number, averageAge: number, ageList: number[],
              numberOfMaleUsersWhoAnswered: number, numberOfFemaleUsersWhoAnswered: number, countriesList: string[],
              numberOfDifferentCountries: number, numberOfInvitationsSent: number, percentOfInvitationsAccepted: number,
              listOfFeedbacks: string[][], numberOfPositiveFeedbacks: number, numberOfNegativeFeedbacks: number, numberOfMixedFeedbacks: number,
              numberOfNeutralFeedbacks: number, ratings: number[], averageRating: number) {
    this.id = id;
    this.survey = survey;
    this.numberOfAnswers = numberOfAnswers;
    this.percentOfUsersWhoAnswered = percentOfUsersWhoAnswered;
    this.averageAge = averageAge;
    this.ageList = ageList;
    this.numberOfMaleUsersWhoAnswered = numberOfMaleUsersWhoAnswered;
    this.numberOfFemaleUsersWhoAnswered = numberOfFemaleUsersWhoAnswered;
    this.countriesList = countriesList;
    this.numberOfDifferentCountries = numberOfDifferentCountries;
    this.numberOfInvitationsSent = numberOfInvitationsSent;
    this.percentOfInvitationsAccepted = percentOfInvitationsAccepted;
    this.listOfFeedbacks = listOfFeedbacks;
    this.numberOfPositiveFeedbacks = numberOfPositiveFeedbacks;
    this.numberOfNegativeFeedbacks = numberOfNegativeFeedbacks;
    this.numberOfMixedFeedbacks = numberOfMixedFeedbacks;
    this.numberOfNeutralFeedbacks = numberOfNeutralFeedbacks;
    this.ratings = ratings;
    this.averageRating = averageRating;
  }
}
